sql/parser/builtins.go

```diff
@@ -443,6 +424,190 @@ var builtins = map[string][]builtin{
 	"sum": aggregateImpls(intType, floatType),
+
+	// Math functions
+
+	"abs": {
+		builtin{
+			returnType: DummyFloat,
+			types:      typeList{floatType},
+			fn: func(args DTuple) (Datum, error) {
+				return DFloat(math.Abs(float64(args[0].(DFloat)))), nil
+			},
+		},
+		builtin{
+			returnType: DummyInt,
+			types:      typeList{intType},
+			fn: func(args DTuple) (Datum, error) {
+				x := args[0].(DInt)
+				switch {
+				case x == math.MinInt64:
+					return DNull, errAbsOfMinInt64
+				case x < 0:
+					return -x, nil
+				}
+				return x, nil
+			},
+		},
+	},
+
+	"acos": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Acos(x)), nil
+		}),
+	},
+
+	"asin": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Asin(x)), nil
+		}),
+	},
+
+	"atan": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Atan(x)), nil
+		}),
+	},
+
+	"atan2": {
+		floatBuiltin2(func(x, y float64) (Datum, error) {
+			return DFloat(math.Atan2(x, y)), nil
+		}),
+	},
+
+	"ceil":    {ceilImpl},
+	"ceiling": {ceilImpl},
+
+	"cos": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Cos(x)), nil
+		}),
+	},
+
+	"degrees": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(180.0 * x / math.Pi), nil
+		}),
+	},
+
+	"div": {
+		floatBuiltin2(func(x, y float64) (Datum, error) {
+			return DFloat(x / y), nil
+		}),
+	},
+
+	"exp": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Exp(x)), nil
+		}),
+	},
+
+	"floor": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Floor(x)), nil
+		}),
+	},
+
+	"ln": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Log(x)), nil
+		}),
+	},
+
+	"log": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Log10(x)), nil
+		}),
+	},
+
+	"mod": {
+		floatBuiltin2(func(x, y float64) (Datum, error) {
+			return DFloat(math.Mod(x, y)), nil
+		}),
+		builtin{
+			returnType: DummyInt,
+			types:      typeList{intType, intType},
+			fn: func(args DTuple) (Datum, error) {
+				y := args[1].(DInt)
+				if y == 0 {
+					return DNull, errDivisionByZero
+				}
+				x := args[0].(DInt)
+				return DInt(x - (x/y)*y), nil
+			},
+		},
+	},
+
+	"pi": {
+		builtin{
+			returnType: DummyFloat,
+			types:      typeList{},
+			fn: func(args DTuple) (Datum, error) {
+				return DFloat(math.Pi), nil
+			},
+		},
+	},
+
+	"pow":   {powImpl},
+	"power": {powImpl},
+
+	"radians": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(x * math.Pi / 180.0), nil
+		}),
+	},
+
+	// TODO(thschroeter): implement round
+
+	"sin": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			return DFloat(math.Sin(x)), nil
+		}),
+	},
+
+	"sign": {
+		floatBuiltin1(func(x float64) (Datum, error) {
+			switch {
+			case x < 0:
+				return DFloat(-1), nil
+			case x == 0:
+				return DFloat(0), nil
+			}
+			return DFloat(1), nil
```

Comment:
Yes, I thought it's probably best to follow postgres behavior if it's easy to do. Before I was sticking closer to the math functions Go provides. I guess, it would be good to compare the results of the math builtins against postgres and document differences.